webapp/channels/src/components/admin_console/cluster_settings.tsx

```diff
@@ -3,36 +3,43 @@
 
 import React from 'react';
 import {FormattedMessage} from 'react-intl';
+import {AdminConfig, ClientLicense, ClusterSettings as ClusterSettingsType} from '@mattermost/types/config';
 
 import {Client4} from 'mattermost-redux/client';
 
 import * as Utils from 'utils/utils';
 
 import WarningIcon from 'components/widgets/icons/fa_warning_icon';
+import {DeepPartial} from '@mattermost/types/utilities';
 
 import ExternalLink from 'components/external_link';
 
-import AdminSettings from './admin_settings';
+import AdminSettings, {BaseProps, BaseState} from './admin_settings';
 import BooleanSetting from './boolean_setting';
 import ClusterTableContainer from './cluster_table_container.jsx';
 import SettingsGroup from './settings_group';
 import TextSetting from './text_setting';
 
-export default class ClusterSettings extends AdminSettings {
-    getConfigFromState = (config) => {
+export type ClusterSettingsProps = BaseProps & {
+    license: ClientLicense;
+};
+type State = BaseState & ClusterSettingsType;
+
+export default class ClusterSettings extends AdminSettings<ClusterSettingsProps, State> {
+    getConfigFromState = (config: AdminConfig) => {
         config.ClusterSettings.Enable = this.state.Enable;
         config.ClusterSettings.ClusterName = this.state.ClusterName;
         config.ClusterSettings.OverrideHostname = this.state.OverrideHostname;
         config.ClusterSettings.UseIPAddress = this.state.UseIPAddress;
         config.ClusterSettings.EnableExperimentalGossipEncryption = this.state.EnableExperimentalGossipEncryption;
         config.ClusterSettings.EnableGossipCompression = this.state.EnableGossipCompression;
-        config.ClusterSettings.GossipPort = this.parseIntNonZero(this.state.GossipPort, 8074);
-        config.ClusterSettings.StreamingPort = this.parseIntNonZero(this.state.StreamingPort, 8075);
+        config.ClusterSettings.GossipPort = this.parseIntNonZero(String(this.state.GossipPort), 8074);
+        config.ClusterSettings.StreamingPort = this.parseIntNonZero(String(this.state.StreamingPort), 8075);
         return config;
     };
 
-    getStateFromConfig(config) {
-        const settings = config.ClusterSettings;
+    getStateFromConfig(config: DeepPartial<AdminConfig>): Partial<State> {
```

Comment:
this can be just Props['config'] and remove the explicit return type

Reply:
I dont understand why the return type is not necessary here